test/core/event_engine/fuzzing_event_engine/fuzzing_event_engine.h

```diff
@@ -48,62 +57,196 @@ class FuzzingEventEngine : public EventEngine {
-  TaskHandle RunAfter(Duration when, Closure* closure) override;
-  TaskHandle RunAfter(Duration when,
-                      absl::AnyInvocable<void()> closure) override;
-  bool Cancel(TaskHandle handle) override;
+  void Run(Closure* closure) ABSL_LOCKS_EXCLUDED(mu_) override;
+  void Run(absl::AnyInvocable<void()> closure)
+      ABSL_LOCKS_EXCLUDED(mu_) override;
+  TaskHandle RunAfter(Duration when, Closure* closure)
+      ABSL_LOCKS_EXCLUDED(mu_) override;
+  TaskHandle RunAfter(Duration when, absl::AnyInvocable<void()> closure)
+      ABSL_LOCKS_EXCLUDED(mu_) override;
+  bool Cancel(TaskHandle handle) ABSL_LOCKS_EXCLUDED(mu_) override;
 
   using Time = std::chrono::time_point<FuzzingEventEngine, Duration>;
 
   Time Now() ABSL_LOCKS_EXCLUDED(mu_);
 
-  static void SetGlobalNowImplEngine(FuzzingEventEngine* engine)
-      ABSL_LOCKS_EXCLUDED(mu_);
-  static void UnsetGlobalNowImplEngine(FuzzingEventEngine* engine)
-      ABSL_LOCKS_EXCLUDED(mu_);
+  // Clear any global hooks installed by this event engine. Call prior to
+  // destruction to ensure no overlap between tests if constructing/destructing
+  // each test.
+  void UnsetGlobalHooks() ABSL_LOCKS_EXCLUDED(mu_);
 
  private:
+  // One pending task to be run.
   struct Task {
     Task(intptr_t id, absl::AnyInvocable<void()> closure)
         : id(id), closure(std::move(closure)) {}
     intptr_t id;
     absl::AnyInvocable<void()> closure;
   };
 
+  // Per listener information.
+  // We keep a shared_ptr to this, one reference held by the FuzzingListener
+  // Listener implementation, and one reference in the event engine state, so it
+  // may be iterated through and inspected - principally to discover the ports
+  // on which this listener is listening.
+  struct ListenerInfo {
+    ListenerInfo(
+        Listener::AcceptCallback on_accept,
+        absl::AnyInvocable<void(absl::Status)> on_shutdown,
+        std::unique_ptr<MemoryAllocatorFactory> memory_allocator_factory)
+        : on_accept(std::move(on_accept)),
+          on_shutdown(std::move(on_shutdown)),
+          memory_allocator_factory(std::move(memory_allocator_factory)),
+          started(false) {}
+    ~ListenerInfo() ABSL_LOCKS_EXCLUDED(mu_);
+    // The callback to invoke when a new connection is accepted.
+    Listener::AcceptCallback on_accept;
+    // The callback to invoke when the listener is shut down.
+    absl::AnyInvocable<void(absl::Status)> on_shutdown;
+    // The memory allocator factory to use for this listener.
+    const std::unique_ptr<MemoryAllocatorFactory> memory_allocator_factory;
+    // The ports on which this listener is listening.
+    std::vector<int> ports ABSL_GUARDED_BY(mu_);
+    // Has start been called on the listener?
+    // Used to emulate the Bind/Start semantics demanded by the API.
+    bool started ABSL_GUARDED_BY(mu_);
+    // The status to return via on_shutdown.
+    absl::Status shutdown_status ABSL_GUARDED_BY(mu_) = absl::OkStatus();
```

Comment:
Initialization here is unnecessary, the default constructor is `absl::OkStatus()`.

Reply:
fixed